lib/auth/webauthncli/fido2.go

```diff
@@ -131,28 +131,29 @@ func fido2Login(
 	var usedAppID bool
 
 	pathToRPID := &sync.Map{} // map[string]string
-	filter := func(dev FIDODevice, info *deviceInfo) (bool, error) {
+	// TODO (codingllama): not sure if capability is best name, maybe you got better suggestion.
+	deviceCapabilityCheck := func(dev FIDODevice, info *deviceInfo) error {
 		switch {
 		case uv && !info.uvCapable():
 			log.Debugf("FIDO2: Device %v: filtered due to lack of UV", info.path)
-			return false, nil
+			// TODO (codingllama): shopuld we suggest setting pin here if it has uv option present but set to false?
+			return trace.BadParameter("Device not capable due to lack of UV")
 		case passwordless && !info.rk:
 			log.Debugf("FIDO2: Device %v: filtered due to lack of RK", info.path)
-			return false, nil
+			return trace.BadParameter("Device not capable due to lack of resident keys")
```

Comment:
Maybe:

```suggestion
			return trace.BadParameter("device does not support passwordless (lacking resident key feature)")
```

UX could help :)